include/pybind11/pytypes.h

```diff
@@ -2226,6 +2226,16 @@ class kwargs : public dict {
     PYBIND11_OBJECT_DEFAULT(kwargs, dict, PyDict_Check)
 };
 
+template <typename T>
```

Comment:
Could you please add a terse comment here, to explain what this is for. (And/or maybe point to this PR.)